src/mysky/encrypted_files.ts

```diff
@@ -0,0 +1,330 @@
+ * at least 5% of the size of the file.
+ *
+ * For example, a 1 kib encrypted file would be padded to 4 kib, a 5 kib file
+ * would be padded to 8 kib, and a 105 kib file would be padded to 112 kib.
+ * Below is a short table of valid file sizes:
+ *
+ * ```
+ *   4 KiB      8 KiB     12 KiB     16 KiB     20 KiB
+ *  24 KiB     28 KiB     32 KiB     36 KiB     40 KiB
+ *  44 KiB     48 KiB     52 KiB     56 KiB     60 KiB
+ *  64 KiB     68 KiB     72 KiB     76 KiB     80 KiB
+ *
+ *  88 KiB     96 KiB    104 KiB    112 KiB    120 KiB
+ * 128 KiB    136 KiB    144 KiB    152 KiB    160 KiB
+ *
+ * 176 KiB    192 Kib    208 KiB    224 KiB    240 KiB
+ * 256 KiB    272 KiB    288 KiB    304 KiB    320 KiB
+ *
+ * 352 KiB    ... etc
+ * ```
+ *
+ * Note that the first 20 valid sizes are all a multiple of 4 KiB, the next 10
+ * are a multiple of 8 KiB, and each 10 after that the multiple doubles. We use
+ * this method of padding files to prevent an adversary from guessing the
+ * contents or structure of the file based on its size.
+ *
+ * @param initialSize - The size of the file.
+ * @returns - The final size, padded to a pad block.
+ * @throws - Will throw if the size would overflow the JS number type.
+ */
+export function padFileSize(initialSize: number): number {
+  const kib = 1 << 10;
+  for (let n = 0; ; n++) {
+    // Prevent overflow. Max JS number size is 2^53-1.
+    if (n >= 53) {
+      throw new Error("Could not pad file size, overflow detected.");
+    }
+    if (initialSize <= (1 << n) * 80 * kib) {
+      const paddingBlock = (1 << n) * 4 * kib;
+      let finalSize = initialSize;
+      if (finalSize % paddingBlock != 0) {
+        finalSize = initialSize - (initialSize % paddingBlock) + paddingBlock;
+      }
+      return finalSize;
+    }
+  }
+}
+
+/**
+ * Checks if the given size corresponds to the correct padded block.
+ *
+ * @param size - The given file size.
+ * @returns - Whether the size corresponds to a padded block.
+ * @throws - Will throw if the size would overflow the JS number type.
+ */
+export function checkPaddedBlock(size: number): boolean {
+  const kib = 1 << 10;
+  for (let n = 0; ; n++) {
+    // Prevent overflow. Max JS number size is 2^53-1.
+    if (n >= 53) {
+      throw new Error("Could not check padded file size, overflow detected.");
+    }
+    if (size <= (1 << n) * 80 * kib) {
+      const paddingBlock = (1 << n) * 4 * kib;
+      return size % paddingBlock === 0;
+    }
+  }
+}
+
+/**
+ * Decodes the encoded encrypted file metadata.
+ *
+ * @param bytes - The encoded metadata.
+ * @returns - The decoded metadata.
+ * @throws - Will throw if the given bytes are of the wrong length.
+ */
+function decodeEncryptedFileMetadata(bytes: Uint8Array): EncryptedFileMetadata {
+  // Ensure input is correct length.
+  validateUint8ArrayLen("bytes", bytes, "encrypted file metadata bytes", ENCRYPTION_HIDDEN_FIELD_METADATA_LENGTH);
+
+  const version = bytes[0];
+
+  return {
+    version,
+  };
```

Comment:
```suggestion
  return { version: bytes[0] };
```

Reply:
Not fixing since lints/prettier didn't catch it.